File: 1328-break-a-palindrome/1328-break-a-palindrome.java
class Solution {
    public String breakPalindrome(String palindrome) {
        if(palindrome.length() == 1) return "";
        char[] arr = palindrome.toCharArray();
        int i = 0;
        while(i < arr.length/2 && arr[i] == 'a') ++i;
        if(i == arr.length/2){
            arr[arr.length-1] = 'b';
        } else
            arr[i] = 'a';
        return new String(arr);
    }
}

File: 1328-break-a-palindrome/1328-break-a-palindrome.py
class Solution:
    def breakPalindrome(self, palindrome: str) -> str:
        if len(palindrome) == 1:
            return ""
        
        c_arr = list(palindrome)
        i = 0
        
        while i < len(palindrome) // 2 and c_arr[i] == 'a':
            i += 1
        
        if i < len(palindrome) // 2:
            c_arr[i] = 'a'
        else:
            c_arr[-1] = 'b'
        
        return ''.join(c_arr)